ignite/distributed/comp_models/native.py

```diff
@@ -419,14 +419,16 @@ def spawn(  # type: ignore[override]
             "OR": dist.ReduceOp.BOR,
         }
 
-        def _do_all_reduce(self, tensor: torch.Tensor, op: str = "SUM") -> torch.Tensor:
+        def _do_all_reduce(
+            self, tensor: torch.Tensor, op: str = "SUM", group: Optional[Union[Any, List[int]]] = None
+        ) -> torch.Tensor:
             if op not in self._reduce_op_map:
                 raise ValueError(f"Unsupported reduction operation: '{op}'")
             reduce_op = self._reduce_op_map[op]
-            dist.all_reduce(tensor, reduce_op)
+            dist.all_reduce(tensor, reduce_op, group=group)
```

Comment:
There is something missing here I think as if group is a list of ints, `dist.all_reduce` would complain about wrong input type ?

Reply:
So we make each group from `new_group` in `ignite.distributed.utils.py`
Should I modify to prevent wrong input type from each model's `_do_all_reduce` ?

Reply:
Yes, we should check for input type.

> So we make each group from new_group in ignite.distributed.utils.py

And if group is List[int], there is no `new_group` call here... So, it would fail